proposals/cleanup.md

```diff
@@ -233,6 +233,87 @@ Kyverno could alternatively leverage a CronJob resource to perform the deletions
 | CronJob will do both: schedule the deletion and execute the deletion. | A CronJob created through the cleanup rule should be ignored by other rules(validate and mutate). For example, A validate policy should not block the creation of that CronJob. This can cause some complexities in the approach. We can label these CronJobs and ignore them while applying other policies. |
 | The CronJobs can be easily removed from the cluster after the completion by setting their ownerReference to the resource it was created to delete. |  |
 
+## Finalized Implementation
+
+Finalized implementation is **Implementation 2 (reuse the Kubernetes CronJob resource)**. So, we will create new CRDs **CleanupPolicy** and **ClusterCleanupPolicy** for handling resource cleanup. 
+
+CleanupPolicy CRD manfiests example:
+```
+apiVersion: kyverno.io/v1alpha1
+kind: CleanupPolicy
+metadata:
+  name: cleanuppolicy
+  namespace: default
+spec:
+  match:
+    any:
+    - resources:
+        kinds:
+          - Pod
+  conditions:
+    any:
+    - key: "{{ request.name }}"
+      operator: Equals
+      value: example
+  schedule: "* * * * *"
+```
+A cleanup controller will reconcile these CRDs and create CronJobs which will execute commands to delete the specified resources. For implementing cleanup controller we have 2 options:
+
+
+### 1. Creating CronJobs per resource.
+A cleanup controller will be creating a CronJob for each of the matching resources. CronJob will run the kubectl command to delete these resources. CronJob will run according to the schedule specified in the cleanupPolicy. Creation and updation of a new cleanup policy will trigger the reconciliation and the reconciliation logic will get all the matching resources and will create a CronJob per resource which will have info and commands to delete the matching resource.
+
+
+
+**Flow:**
+
+1. A new policy is added to the cluster. 
+2. The cleanup policy object is added to the controller queue with the help of CleanupPolicy and ClusterCleanupPolicy informers.
+3. The controller then gets the policy.
+4. Get the matching resources for that rule/policy.
+5. Create CronJob for scheduling the deletion for each resource.
+6. Kyverno’s Service account is used in CronJob.
+7. OwnerReference of the CronJob is set to the trigger resource, so when the matching resource is deleted, the CJ is also deleted.
```

Comment:
I would expect the end-to-end reconciliation logic to be added to the Flow section but failed to envision that way.